### ISSUE TITLE
Default Installation is causing errors

### ISSUE THREAD

**Issue by quasar66** [tensorflow/tensorflow#45671]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 Pro
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Nope
- TensorFlow installed from (source or binary): Source
- TensorFlow version: 2.4 released 12/14/2020
- Python version: 3.8.6 installed directly from the official site
- Installed using virtualenv? pip? conda?: Nope
- Bazel version (if compiling from source): Nope
- GCC/Compiler version (if compiling from source): Nope
- CUDA/cuDNN version: both 11.1/8.0.5 and 11.0/8.0.2
- GPU model and memory: GEForce GTX1650 with 2 GB RAM

**Describe the problem**
Unable to do a default installation from source
If I use CUDA 11.0 with cuDNN 8.0.2.39 for 11.0, the error is
_E tensorflow/stream_executor/cuda/cuda_blas.cc:226] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED_

Alternatively, if I use CUDA 11.1 with cuDNN 8.0.5.39 for 11.1, the error is
_Could not load dynamic library 'cusolver64_10.dll'; dlerror: cusolver64_10.dll not found
Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices..._

**Provide the exact sequence of commands / steps that you executed before running into the problem**
pip install tensorflow

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by amahendrakar** [comment]
@quasar66,
> Alternatively, if I use CUDA 11.1 with cuDNN 8.0.5.39 for 11.1, the error is
> _Could not load dynamic library 'cusolver64_10.dll'; dlerror: cusolver64_10.dll not found_

TensorFlow v2.4 and TF-nightly are built and tested against CUDA 11. Hence, could you please install CUDA 11 instead of CUDA 11.1 to resolve this issue. 

Also, go through these similar issues [#44567](https://github.com/tensorflow/tensorflow/issues/44567#issuecomment-729504167), [#45258](https://github.com/tensorflow/tensorflow/issues/45258#issuecomment-739871133) for more information.

**Comment by amahendrakar** [comment]
> If I use CUDA 11.0 with cuDNN 8.0.2.39 for 11.0, the error is
> _E tensorflow/stream_executor/cuda/cuda_blas.cc:226] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED_

Regarding the `CUBLAS_STATUS_ALLOC_FAILED_` error, could you please provide the exact sequence of commands and code that you executed before running into the error, so that we can look into this. Thanks!

**Comment by quasar66** [comment]
HI..
The following is a toy code that will reflect the CUBLAS_STATUS_ALLOC_FAILED problem:

import numpy as np
import tensorflow as tf
from sklearn.datasets import load_sample_image

china = load_sample_image("china.jpg") / 255
flower = load_sample_image("flower.jpg") / 255
images = np.array([china, flower])
batch_size, height, width, channels = images.shape

filters = np.zeros(shape=(7, 7, channels, 2), dtype=np.float32)
filters[:, 3, :, 0] = 1  # vertical line
filters[3, :, :, 1] = 1  # horizontal line

outputs = tf.nn.conv2d(images, filters, strides=1, padding="SAME")


The corresponding warning and error messages are as below:

_2020-12-15 09:28:45.403935: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudart64_110.dll
2020-12-15 09:28:46.956383: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
2020-12-15 09:28:46.957868: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library nvcuda.dll
2020-12-15 09:28:48.041053: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce GTX 1650 computeCapability: 7.5
coreClock: 1.56GHz coreCount: 16 deviceMemorySize: 4.00GiB deviceMemoryBandwidth: 119.24GiB/s
2020-12-15 09:28:48.041363: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudart64_110.dll
2020-12-15 09:28:48.048683: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublas64_11.dll
2020-12-15 09:28:48.048843: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublasLt64_11.dll
2020-12-15 09:28:48.052785: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cufft64_10.dll
2020-12-15 09:28:48.054430: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library curand64_10.dll
2020-12-15 09:28:48.062725: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusolver64_10.dll
2020-12-15 09:28:48.065722: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusparse64_11.dll
2020-12-15 09:28:48.066828: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudnn64_8.dll
2020-12-15 09:28:48.067032: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0
2020-12-15 09:28:48.611438: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2020-12-15 09:28:49.115463: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:01:00.0 name: GeForce GTX 1650 computeCapability: 7.5
coreClock: 1.56GHz coreCount: 16 deviceMemorySize: 4.00GiB deviceMemoryBandwidth: 119.24GiB/s
2020-12-15 09:28:49.115759: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudart64_110.dll
2020-12-15 09:28:49.115905: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublas64_11.dll
2020-12-15 09:28:49.116066: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublasLt64_11.dll
2020-12-15 09:28:49.116219: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cufft64_10.dll
2020-12-15 09:28:49.116363: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library curand64_10.dll
2020-12-15 09:28:49.116504: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusolver64_10.dll
2020-12-15 09:28:49.116651: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusparse64_11.dll
2020-12-15 09:28:49.116834: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudnn64_8.dll
2020-12-15 09:28:49.117020: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0
2020-12-15 09:28:51.110467: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-12-15 09:28:51.110634: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      0 
2020-12-15 09:28:51.110740: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 0:   N 
2020-12-15 09:28:51.112568: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2903 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1650, pci bus id: 0000:01:00.0, compute capability: 7.5)
2020-12-15 09:28:51.117135: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
2020-12-15 09:28:51.254734: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudnn64_8.dll
2020-12-15 09:28:53.830147: I tensorflow/core/platform/windows/subprocess.cc:308] SubProcess ended with return code: 0

2020-12-15 09:28:53.880606: I tensorflow/core/platform/windows/subprocess.cc:308] SubProcess ended with return code: 0

2020-12-15 09:28:54.168327: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublas64_11.dll
2020-12-15 09:28:55.246945: **E tensorflow/stream_executor/cuda/cuda_blas.cc:226] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED**

Process finished with exit code 0_

**Comment by amahendrakar** [comment]
@quasar66,
Please try limiting the GPU memory growth using any one of the methods listed in [this guide](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth) and let us know if it helps.

Also, please take a look at [this comment](https://github.com/tensorflow/tensorflow/issues/7072#issuecomment-687649994) from similar issue #7072 and check if it works. Thanks!

**Comment by quasar66** [comment]
Thanks !!
The comment link you provided worked...
Am closing the issue.
